src/approximations/nystrom.jl

```diff
@@ -0,0 +1,104 @@
+# Following the algorithm by William and Seeger, 2001
+# Cs is equivalent to X_mm and C to X_mn
+
+function sampleindex(X::AbstractMatrix, r::Real; obsdim::Integer=defaultobs)
+    0 < r <= 1 || throw(ArgumentError("Sample rate `r` must be in range (0,1]"))
+    n = size(X, obsdim)
+    m = ceil(Int, n*r)
+    S = StatsBase.sample(collect(1:n), m; replace=false, ordered=true)
+    return S
+end
+
+function nystrom_sample(k::Kernel, X::AbstractMatrix, S::Vector{<:Integer}; obsdim::Integer=defaultobs)
+    obsdim ∈ [1, 2] || throw(ArgumentError("`obsdim` should be 1 or 2 (see docs of kernelmatrix))"))
+    Xₘ = obsdim == 1 ? getindex(X, S, :) : getindex(X, :, S)
```

Comment:
That's my personal preference, but I think it would be easier to read if you write
```suggestion
    Xₘ = obsdim == 1 ? X[S, :] : X[:, S]
```
and
```julia
    Cs = C[:, S]
```
two lines below.